server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderUnitTests.java

```diff
@@ -472,22 +477,113 @@ public void testShardSizeAndRelocatingSize() {
+        Index anotherIndex = new Index(anotherIndexName, "5678");
+        Metadata metadata = Metadata.builder()
+            .put(
+                IndexMetadata.builder(mainIndexName)
+                    .settings(
+                        settings(Version.CURRENT).put("index.uuid", "1234")
+                            .put(INDEX_STORE_TYPE_SETTING.getKey(), SEARCHABLE_SNAPSHOT_STORE_TYPE)
+                    )
+                    .numberOfShards(3)
+                    .numberOfReplicas(1)
+            )
+            .put(
+                IndexMetadata.builder(anotherIndexName)
+                    .settings(settings(Version.CURRENT).put("index.uuid", "5678"))
+                    .numberOfShards(1)
+                    .numberOfReplicas(1)
+            )
+            .build();
+        String nodeId = "node1";
+        String anotherNodeId = "another_node";
+
+        List<ShardRouting> shards = new ArrayList<>();
+        int anotherNodeShardCounter = 0;
+        for (int i = 1; i <= 3; i++) {
+            int expectedSize = 10 * i;
+            shards.add(createShard(index, nodeId, i - 1, expectedSize));
+            // randomly add shard for non-searchable snapshot index
+            if (randomBoolean()) {
+                shards.add(createShard(anotherIndex, anotherNodeId, anotherNodeShardCounter++, expectedSize));
+            }
+        }
+
+        DiscoveryNode node = new DiscoveryNode(nodeId, buildNewFakeTransportAddress(), emptyMap(), emptySet(), Version.CURRENT);
+        DiscoveryNode anotherNode = new DiscoveryNode(
+            anotherNodeId,
+            buildNewFakeTransportAddress(),
+            emptyMap(),
+            emptySet(),
+            Version.CURRENT
+        );
+        ClusterState clusterState = ClusterState.builder(ClusterName.CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY))
+            .metadata(metadata)
+            .routingTable(
+                RoutingTable.builder()
+                    .add(
+                        shards.stream()
+                            .filter(s -> s.getIndexName().equals(mainIndexName))
+                            .reduce(IndexRoutingTable.builder(index), IndexRoutingTable.Builder::addShard, (a, b) -> a)
+                    )
+                    .add(
+                        shards.stream()
+                            .filter(s -> s.getIndexName().equals(anotherIndexName))
+                            .reduce(IndexRoutingTable.builder(anotherIndex), IndexRoutingTable.Builder::addShard, (a, b) -> a)
+                    )
+                    .build()
+            )
+            .nodes(DiscoveryNodes.builder().add(node).add(anotherNode).build())
+            .build();
+        RoutingAllocation allocation = new RoutingAllocation(
+            null,
+            clusterState,
+            new DevNullClusterInfo(Map.of(), Map.of(), Map.of()),
+            null,
+            0
+        );
+        long sizeOfRelocatingShards = sizeOfUnaccountedShards(
```

Comment:
can we add some initializing searchable snapshot shards randomly here to ensure that those are accounted as well?